### PR TITLE
Feature: support multiple package managers

### LINKED ISSUE #173: Add support for the `workspace:` protocol
I'm using a [PNPM workspace](https://pnpm.io/workspaces) for my project but when I use the `workspace:` protocol with the `pnpm attw --pack .` command I get the following error:
```xl
error while checking file:
Command failed: npm pack
```
Running `npm pack` manually reveals the following in the debug file:
```xl
14 verbose stack TypeError: Cannot set properties of null (setting 'peer')
14 verbose stack     at visit (/opt/homebrew/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/calc-dep-flags.js:101:54)
14 verbose stack     at visitNode (/opt/homebrew/lib/node_modules/npm/node_modules/treeverse/lib/depth-descent.js:58:25)
14 verbose stack     at next (/opt/homebrew/lib/node_modules/npm/node_modules/treeverse/lib/depth-descent.js:44:19)
14 verbose stack     at depth (/opt/homebrew/lib/node_modules/npm/node_modules/treeverse/lib/depth-descent.js:83:10)
14 verbose stack     at depth (/opt/homebrew/lib/node_modules/npm/node_modules/treeverse/lib/depth.js:27:12)
14 verbose stack     at unsetFlag (/opt/homebrew/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/calc-dep-flags.js:96:5)
14 verbose stack     at /opt/homebrew/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/calc-dep-flags.js:63:7
14 verbose stack     at CIMap.forEach (<anonymous>)
14 verbose stack     at calcDepFlagsStep (/opt/homebrew/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/calc-dep-flags.js:41:17)
14 verbose stack     at calcDepFlagsStep (/opt/homebrew/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/calc-dep-flags.js:38:12)
15 error Cannot set properties of null (setting 'peer')
```
I currently use `pnpm attw $(pnpm pack)` as workaround but it would be nice if the `--pack` option used the same package manager.
This could be detected automatically with a package like [detect-package-manager](https://www.npmjs.com/package/detect-package-manager) or passed manually as an option.
[Yarn](https://yarnpkg.com/features/workspaces) seems to use this protocol too for workspaces.

### PR DESCRIPTION
**Pull request by goestav**
This pull request adds support for other package managers like [pnpm](https://pnpm.io/) and [yarn](https://yarnpkg.com/).

Closes #173

### CODE REVIEW

File: packages/cli/tsconfig.json [hunk 1]
```diff
@@ -5,7 +5,8 @@
     "rootDir": "src",
     "types": ["ts-expose-internals-conditionally", "node"],
     "outDir": "./dist",
-    "sourceMap": true
+    "sourceMap": true,
+    "allowSyntheticDefaultImports": true
```
Comment (andrewbranch) [inline packages/cli/tsconfig.json:9]: This is implied by `--module nodenext`

```suggestion
    "sourceMap": true
```

**Review by andrewbranch — CHANGES_REQUESTED** [review]
Thank you! All looks good except for the config change.

File: packages/cli/tsconfig.json (hunk 1 above)
Comment (goestav) [reply packages/cli/tsconfig.json]: Strange, I remember getting an error or warning without that option but now it seems fine. I removed it in b8d7fa0.

**Comment by goestav** [comment]
Hello @andrewbranch, you're correct about pnpm not supporting custom filenames for tarballs.
I did mention that in the comment [here](https://github.com/goestav/arethetypeswrong.github.io/blob/cbcca7ebbe9ab147efa84de0692b34c1f28dd561/packages/cli/src/utils.ts#L8) but maybe I should've made that clear in the merge request description as well. Yarn [does support it](https://yarnpkg.com/cli/pack#options) however.

What do you think about an option that lets you specify the package manager manually for the pack command? Like for example: `attw --pack . --package-manager pnpm` (or `attw --pack . -m pnpm` for short).
The default value should be `npm` so it doesn't break the workflow of others.

**Pull request merged**